doc/dev/cephadm/developing-cephadm.rst

```diff
@@ -256,3 +226,105 @@ When completed, you'll see::
   "Localized bundle generation complete."
 
 Then you can reload your Dashboard browser tab.
+
+Cephadm DiD (Docker in Docker) box development environment
+==========================================================
+
+As kcli has a long startup time, we created an alternative which is faster using
+Docker inside Docker. This approach has its downsides too as we have to
+simulate the creation of osds and addition of devices with loopback devices.
+
+Cephadm's DiD environment is a command which requires little to setup. The setup
+requires you to get the required docker images for what we call boxes and ceph.
+A box is the first layer of docker containers which can be either a seed or a
+host. A seed is the main box which holds cephadm and where you bootstrap the
+cluster. On the other hand, you have hosts with an ssh server setup so you can
+add those hosts to the cluster. The second layer, managed by cephadm, inside the
+seed box, requires the ceph image.
+
+Requirements
+------------
+
+* `docker-compose <https://docs.docker.com/compose/install/>`_
+* lvm
+
+Setup
+-----
+
+In order to setup Cephadm's box run::
+
+        cd src/cephadm/box
+        ln -s box.py /usr/bin/box
+        sudo box -v cluster setup
+
+After getting all needed images we can run::
+
+        sudo box -v cluster start --osds 3 --hosts 3
+
+.. note:: Cluster start will try to setup even if cluster setup was not called.
+.. note:: Osds are created with loopback devices and hence, sudo is needed to
+   create loopback devices capable of holding osds.
+.. note::  Each osd will require 5GiB of space.
+
```

Comment:
It would be great to add a note on how to access the Dashboard once the cluster is up, as the expansion can be done either by CLI or by GUI.